build.gradle

```diff
@@ -549,40 +550,44 @@ check.dependsOn jacocoTestCoverageVerification
-    compile "org.opensearch.client:opensearch-rest-client:${opensearch_version}"
-    compile group: 'com.google.guava', name: 'guava', version:'31.0.1-jre'
-    compile group: 'org.apache.commons', name: 'commons-math3', version: '3.6.1'
-    compile group: 'com.google.code.gson', name: 'gson', version: '2.8.9'
-    compile group: 'com.yahoo.datasketches', name: 'sketches-core', version: '0.13.4'
-    compile group: 'com.yahoo.datasketches', name: 'memory', version: '0.12.2'
-    compile group: 'commons-lang', name: 'commons-lang', version: '2.6'
-    compile group: 'org.apache.commons', name: 'commons-pool2', version: '2.10.0'
+    compileOnly "org.opensearch:opensearch-job-scheduler-spi:2.0.0.0-alpha1-SNAPSHOT"
+    implementation "org.opensearch:common-utils:${common_utils_version}"
+    implementation "org.opensearch.client:opensearch-rest-client:${opensearch_version}"
+    implementation group: 'com.google.guava', name: 'guava', version:'31.0.1-jre'
+    implementation group: 'com.google.guava', name: 'failureaccess', version:'1.0.1'
+    implementation group: 'org.javassist', name: 'javassist', version:'3.28.0-GA'
+    implementation group: 'org.apache.commons', name: 'commons-math3', version: '3.6.1'
+    implementation group: 'com.google.code.gson', name: 'gson', version: '2.8.9'
+    implementation group: 'com.yahoo.datasketches', name: 'sketches-core', version: '0.13.4'
+    implementation group: 'com.yahoo.datasketches', name: 'memory', version: '0.12.2'
+    implementation group: 'commons-lang', name: 'commons-lang', version: '2.6'
+    implementation group: 'org.apache.commons', name: 'commons-pool2', version: '2.10.0'
 
     // force Jackson version to avoid version conflict issue
     implementation 'software.amazon.randomcutforest:randomcutforest-serialization:2.0.1'
     implementation "com.fasterxml.jackson.core:jackson-core:2.12.6"
     implementation "com.fasterxml.jackson.core:jackson-databind:2.12.6"
     implementation "com.fasterxml.jackson.core:jackson-annotations:2.12.6"
-    compile files('lib/randomcutforest-parkservices-2.0.1.jar')
-    compile files('lib/randomcutforest-core-2.0.1.jar')
+    implementation files('lib/randomcutforest-parkservices-2.0.1.jar')
+    implementation files('lib/randomcutforest-core-2.0.1.jar')
 
     // used for serializing/deserializing rcf models.
-    compile group: 'io.protostuff', name: 'protostuff-core', version: '1.8.0'
-    compile group: 'io.protostuff', name: 'protostuff-runtime', version: '1.8.0'
-    compile group: 'org.apache.commons', name: 'commons-lang3', version: '3.12.0'
-
-    compile "org.jacoco:org.jacoco.agent:0.8.5"
-    compile ("org.jacoco:org.jacoco.ant:0.8.5") {
+    implementation group: 'io.protostuff', name: 'protostuff-core', version: '1.8.0'
+    implementation group: 'io.protostuff', name: 'protostuff-runtime', version: '1.8.0'
+    implementation group: 'io.protostuff', name: 'protostuff-api', version: '1.8.0'
```

Comment:
Why add these new dependencies: `protostuff-api` and `protostuff-collectionschema`?

Reply:
These required to be explicitly added as the compilation errors for Linkedbuffer and Schema  were coming from these.
Compile failures : https://paste.amazon.com/show/snghes/1647551203